engines/scumm/input.cpp

```diff
@@ -135,6 +135,14 @@ void ScummEngine::parseEvent(Common::Event event) {
 			_keyPressed = event.kbd;
 		}
 
+		// HACK: Because we use ASCII values here, it's necessary to
+		// remap keypad keys to always have a corresponding ASCII value.
+		// Normally, keypad keys would only have an ASCII value when
+		// NumLock is enabled. This fixes fighting in Indy 3 (Trac #11227)
+		if (_keyPressed.keycode >= Common::KEYCODE_KP0 && _keyPressed.keycode <= Common::KEYCODE_KP9) {
+			_keyPressed.ascii = (_keyPressed.keycode - Common::KEYCODE_KP0) + '0';
+		}
+
 		// FIXME: We are using ASCII values to index the _keyDownMap here,
```

Comment:
Doesn't the same change need to be made for the `EVENT_KEYUP` case as well to reset the `_keyDownMap` to false for the correspond ascii value?

Reply:
OK, done.